processing/src/main/java/io/druid/query/groupby/having/DimensionSelectorHavingSpec.java

```diff
@@ -48,11 +48,7 @@ public DimensionSelectorHavingSpec(
   {
     dimension = Preconditions.checkNotNull(dimName, "Must have attribute 'dimension'");
     this.value = value;
-    if (extractionFn != null) {
-      this.extractionFn = extractionFn;
-    } else {
-      this.extractionFn = new IdentityExtractionFn(); 
-    }
+    this.extractionFn = extractionFn != null ? extractionFn : IdentityExtractionFn.getInstance();
```

Comment:
Why not just create a local `private static final` or make the instance `public static final` and depend on it.  The only reason to have a method is to introduce laziness and there's no laziness required here.